Gems/Atom/RPI/Code/Source/RPI.Public/RPISystem.cpp

```diff
@@ -83,12 +84,17 @@ namespace AZ
             if (m_xrSystem)
             {
                 AZ::RHI::ResultCode resultCode = m_xrSystem->InitInstance();
-                // Fail result code can happen if no compatible device is attached. UnRegister xr system if that happens
-                if (resultCode == AZ::RHI::ResultCode::Fail)
+                // Check if the result code is a Success. Failure can happen if no compatible device is attached.
+                // Check if the active rhi backend is not vulkan
+                // UnRegister xr system if any of the criteria is not met
+                if (resultCode == AZ::RHI::ResultCode::Fail || !AZ::RHI::IsVulkanRHI())
```

Comment:
if resultCode is success, but the RHI is not vulkan, it will throw out an error and simply unregister XR so that it doesnt crash.